pkg/httpclient/client.go

```diff
@@ -89,6 +89,13 @@ func (c *Client) SetHeader(key string, value string) {
 	c.headers[key] = value
 }
 
+func (c *Client) SetQueryParam(key, value string) {
+	if c.queryParams == nil {
+		c.queryParams = make(map[string]string)
+	}
+	c.queryParams[key] = value
+}
```

Comment:
I'm not very happy with this whole implementation, but I wanted to do as little changes as possible, as this is likely a temporary change to vulture.